metadatastore/test/test_sample_data.py

```diff
@@ -1,42 +1,39 @@
 from metadatastore.examples.sample_data.common import apply_deadband, noisy
 from metadatastore.examples.sample_data import (temperature_ramp,
                                                 multisource_event)
-from metadatastore.utils.testing import mds_setup, mds_teardown
+from .utils import mds_setup, mds_teardown
 import numpy as np
 import itertools
-from nose.tools import raises
+import pytest
 
-def setup():
+
+def setup_function(function):
     mds_setup()
 
-def teardown():
+
+def teardown_function(function):
     mds_teardown()
 
-@raises(ValueError)
+
 def test_bad_deadbands():
     # the -1 should throw a ValueError
-    apply_deadband(None, -1)
+    with pytest.raises(ValueError):
+        apply_deadband(None, -1)
 
 
-def _noisy_helper(val):
-    noisy(val)
+@pytest.mark.parametrize('v', [0, np.arange(0, 5)])
+def test_noisy_for_smoke(v):
+    noisy(v)
 
 
-def test_noisy_for_smoke():
-    vals = [0, np.arange(0, 5)]
-    for v in vals:
-        yield _noisy_helper, v
+@pytest.mark.parametrize(
+    'failing_time, mod',
+    itertools.product([-1, 1], [temperature_ramp, multisource_event])
+)
+def test_sleepy_failures(failing_time, mod):
```

Comment:
It is testing to make sure that our data generation code, in this case `temperature_ramp` and `multisource_event` will correctly fail if they are given a sleep time.  I do not care either way if we keep or remove this test.